lib/rgeo/shapefile/reader.rb

```diff
@@ -176,6 +176,7 @@ def self.open(path_, opts_ = {}, &block_)
       # block. You should use Reader::open instead.
 
       def initialize(path_, opts_ = {}) # :nodoc:
+        @opts = opts_
```

Comment:
Since every other option is treated by handling it at that stage, I'd rather:
```suggestion
        @allow_unsafe = opts_[:allow_unsafe_methods]
```